src/visualizer/codeinformation/codeexplaination/CodeExplanation.js

```diff
@@ -1,74 +1,54 @@
 import React, { useEffect, useState } from 'react';
 import './styles.css';
 import explanations from '../explanations/Explanations';
-import { FaCheckCircle, FaTimesCircle } from 'react-icons/fa';
+import PerformanceSection from './PerformanceSection';
 
+/**
+ * CodeExplanation component which encapsulates the details for how the sorting algorithm works,
+ * and the performance of the algorithm.
+ *
+ * @component
+ * @category CodeExplanation
+ * @param {string} algo Current algorithm selected.
+ * @returns {JSX.Element} Code explanation component.
+ */
 const CodeExplanation = ({ algo }) => {
-  const [sort, setSort] = useState(explanations[algo]);
+  // Sort details of the algorithm.
+  const [sortDetails, setSortDetails] = useState(() => explanations[algo]);
 
   useEffect(() => {
-    setSort(explanations[algo]);
+    setSortDetails(explanations[algo]);
   }, [algo]);
 
-  const HeaderOne = () => <header>How {sort.name} Sort works</header>;
-  const Explanation = () => <p style={{ width: `112%` }}>{sort.description}</p>;
-  const HeaderTwo = () => <header>Performance</header>;
-
-  const PerformanceSection = () => {
-    const ListOfComplexities = () => (
-      <div className="text">
-        <p>Worst time complexity</p>
-        <p>Average time complexity</p>
-        <p>Best time complexity</p>
-        <p>Worst space complexity</p>
-      </div>
-    );
-
-    const Complexity = () => (
-      <div className="complexity">
-        <p>O({sort.worstTime})</p>
-        <p>θ({sort.averageTime})</p>
-        <p>Ω({sort.bestTime})</p>
-        <p>O({sort.worstSpace})</p>
-      </div>
-    );
+  /**
+   * First header component for the explanation section.
+   *
+   * @returns {JSX.Element} First header component for the explanation section.
+   */
```

Comment:
I am not sure if the returns statement value-add anything? it could be simplified as 
```
/**
  * First header component for the explanation section.
  */

Reply:
Does not seem to add much, but doing it to follow jsdoc style.